lib/crystalball/map_storage/yaml_storage.rb

```diff
@@ -8,6 +8,41 @@ class MapStorage
     class YAMLStorage
       attr_reader :path
 
+      class << self
+        def load(path)
+          ymls = read_files(path)
+          guard_metadata_consistency(ymls)
+
+          metadata = ymls.first.first
+          cases = ymls.map(&:last).inject(&:merge!)
+
+          Object.const_get(metadata[:type]).new(metadata: metadata, cases: cases)
+        end
+
+        private
+
+        def read_files(path)
+          paths = path.directory? ? path.each_child.select(&:file?) : [path]
+
+          paths.map do |file|
+            metadata, *cases = file.read.split("---\n").reject(&:empty?).map do |yaml|
+              YAML.safe_load(yaml, [Symbol])
+            end
+            cases = cases.inject(&:merge!)
+
+            [metadata, cases]
+          end
+        end
+
+        def guard_metadata_consistency(data)
+          commits = data.map { |d| d.first[:commit] }.uniq
```

Comment:
I think only checking commit and type is not future proof. Instead we should check that every all metadata is same.